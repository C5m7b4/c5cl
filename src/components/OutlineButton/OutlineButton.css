.btn {
  border: 2px solid black;
  background-color: white;
  color: black;
  padding: 14px 28px;
  font-size: 16px;
  cursor: pointer;
  transition: background-color 0.3s ease-in, color 0.3s ease-in;
}

/* green */
.success {
  border-color: #04aa6d;
  color: green;
}

.success:hover {
  background-color: #04aa6d;
  color: white;
}

.success:disabled {
  background-color: gray;
  color: green;
}

.success:active {
  transform: scale(0.96);
}

/* blue */
.info {
  border-color: #2196f3;
  color: dodgerblue;
}

.info:hover {
  background: #2196f3;
  color: white;
}

.info:disabled {
  background-color: gray;
  color: dodgerblue;
}

.info:active {
  transform: scale(0.96);
}

/* Orange */
.warning {
  border-color: #ff9800;
  color: orange;
}

.warning:hover {
  background: #ff9800;
  color: white;
}

.warning:disabled {
  background-color: gray;
  color: orange;
}

.warning:active {
  transform: scale(0.96);
}

/* red */
.danger {
  border-color: #f44336;
  color: red;
}

.danger:hover {
  background-color: #f44336;
  color: white;
}

.danger:disabled {
  background-color: gray;
  color: red;
}

.danger:active {
  transform: scale(0.96);
}

.default {
  border-color: #e7e7e7;
  color: black;
}

.default:hover {
  background-color: #e7e7e7;
}

.default:disabled {
  background-color: gray;
  color: white;
}

.default:active {
  transform: scale(0.96);
}

.dark {
  border-color: #000;
  color: black;
}

.dark:hover {
  background-color: #000;
  color: white;
}

.dark:disabled {
  background-color: gray;
  color: rgb(192, 181, 181);
}

.dark:active {
  transform: scale(0.96);
}
