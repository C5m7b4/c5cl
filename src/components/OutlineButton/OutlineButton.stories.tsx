import React from 'react';
import { Story, Meta } from '@storybook/react';

import OutlineButton, { OutlineButtonProps } from './OutlineButton';

export default {
  title: 'Buttons',
  component: OutlineButton,
} as Meta;

const Template: Story<OutlineButtonProps> = (args) => (
  <OutlineButton {...args} />
);

export const SuccessButtonExample = Template.bind({});
SuccessButtonExample.args = {
  text: 'Success',
  onClick: () => console.log('success was clicked'),
  type: 'success',
};

export const InfoButtonExample = Template.bind({});
InfoButtonExample.args = {
  text: 'Info',
  onClick: () => console.log('info was clicked'),
  type: 'info',
};

export const WarningButtonExample = Template.bind({});
WarningButtonExample.args = {
  text: 'Warning',
  onClick: () => console.log('warning was clicked'),
  type: 'warning',
};

export const DangerButtonExample = Template.bind({});
DangerButtonExample.args = {
  text: 'Danger',
  onClick: () => console.log('danger was clicked'),
  type: 'danger',
};

export const DefaultButtonExample = Template.bind({});
DefaultButtonExample.args = {
  text: 'Default',
  onClick: () => console.log('default was clicked'),
  type: 'default',
};

export const DarkButtonExample = Template.bind({});
DarkButtonExample.args = {
  text: 'Default',
  onClick: () => console.log('default was clicked'),
  type: 'dark',
};
