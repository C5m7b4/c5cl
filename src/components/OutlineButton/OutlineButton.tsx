import React from 'react';
import './OutlineButton.css';

export type OutlineButtonTypes =
  | 'success'
  | 'dark'
  | 'info'
  | 'error'
  | 'warning'
  | 'danger'
  | 'default';

export interface OutlineButtonProps {
  text: string;
  onClick: () => void;
  disabled?: boolean;
  type: OutlineButtonTypes;
  style?: React.CSSProperties;
}

const OutlineButton = (props: OutlineButtonProps) => {
  const { type, text, disabled = false, onClick, style } = props;
  console.log('type', type);

  return (
    <button
      style={style}
      className={`btn ${type}`}
      disabled={disabled}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default OutlineButton;
