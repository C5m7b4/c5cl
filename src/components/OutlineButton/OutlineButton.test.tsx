import React from 'react';
import { render } from '@testing-library/react';

import OutlineButton from './OutlineButton';

describe('OutlineButton', () => {
  test('should render correctly', () => {
    const testFn = jest.fn;
    const { container } = render(
      <OutlineButton text="Hello" type="default" onClick={testFn} />
    );
    expect(container).toMatchSnapshot();
  });
});
