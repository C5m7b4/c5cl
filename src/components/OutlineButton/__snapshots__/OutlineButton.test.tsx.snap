// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton should render correctly 1`] = `
<div>
  <button
    class="btn default"
  >
    Hello
  </button>
</div>
`;
